controllers/placementapi_controller.go

```diff
@@ -349,6 +348,32 @@ func (r *PlacementAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	// all our input checks out so report InputReady
 	instance.Status.Conditions.MarkTrue(condition.InputReadyCondition, condition.InputReadyMessage)
 
+	// ensure MariaDBAccount exists.  This account record may be created by
+	// openstack-operator or the cloud operator up front without a specific
+	// MariaDBDatabase configured yet.   Otherwise, a MariaDBAccount CR is
+	// created here with a generated username as well as a secret with
+	// generated password.   The MariaDBAccount is created without being
+	// yet associated with any MariaDBDatabase.
+	_, _, err = mariadbv1.EnsureMariaDBAccount(
+		ctx, h, instance.Spec.DatabaseAccount,
+		instance.Namespace, false, placement.DatabaseName,
```

Comment:
do we need to set the requireTLS flag based on some input?

Reply:
tls is used as soon the mariadbdatabase reflects it is hosted on a galera supporting tls and the service has a CA to validate it. we could add setting requiredTLS as a follow up. maybe after we switched our default to enable podlevel/internal tls per default?

Reply:
I'm OK about a follow up, just seemed strange that we merged #6 yesterday in placement, and here we explicitly disable i

Reply:
the ensuremariadbaccount is ultimately going to be a receiver of a MariaDBAccount that may be coming from openstack-operator, and the account is not associated with any database as of yet.   at this stage, it's not appropriate to set TLS.     if we are flipping a TLS flag on when we do the ensure MariaDBDatabase phase, then we would want to flip on "require TLS" on the MariaDBAccount as well, and that would be within the mariadbdatabase_funcs.   if that's the use we want?  that is, if DB supports TLS, then all accounts should *require* TLS.  is that a req?

Reply:
> I'm OK about a follow up, just seemed strange that we merged #165 yesterday in placement, and here we explicitly disable it. So I wanted to make sure it is intentional.

right, setting requireTLS == false is not that then no tls is used for db connection. it is just that this db user could use tls and non tls to connect. if tls is used is being controlled by the my.cnf content. 

>  that is, if DB supports TLS, then all accounts should _require_ TLS. is that a req?

I am not sure right now. I can discuss it in our tomorrows tls meeting and get back

Reply:
thanks for the explanation. I defer to @stuggi if this is OK as is.

Reply:
we did not came up with a real preference if "require TLS" should be used as soon the DB supports TLS. since the services will use tls as soon the DB supports it and the service has the CA we could set it, but atm it won't hurt keep it available. I'd personally do it in a follow up, after we tested all the different scenarios, also with adoption in mind. adoption of tlse env is still in early phase.

Reply:
yah i think this should be all followup stuff.  we're just trying to get all the levers available right now (like, every controller gets a distinct DB password, stuff like that :)  )